### PR TITLE
Templates for template authoring: first edition

### PR DESCRIPTION
**Pull request by vlada-shubina**
### Problem
fixes https://github.com/dotnet/templating/issues/2722

### Solution
Adds the template package with templates for template authoring.

TODO:

- [x] post actions asking to fill TODOs
- [ ] validation of the fields: short name, package ID

### Checks:
- [x] Added unit tests - waits for `Verify` tests to be available
- [ ] Added `#nullable enable` to all the modified files [?](https://github.com/dotnet/templating/wiki/Contributing#coding-style) - N/A

### CODE REVIEW

**Comment by JanKrivanek** [comment]
Do we want to add template for host config file?

**Review by JanKrivanek — CHANGES_REQUESTED** [review]
Great addition!! To mee it almost feel this should be part of sdk :-)

I'd just adjust the json template so that it's valid as is

File: template_feed/Microsoft.TemplateEngine.Authoring.Templates/content/TemplateJson/template.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{
+  "$schema": "http://json.schemastore.org/template",
+  "_comment": "See https://github.com/dotnet/templating/wiki/Reference-for-template.json for complete configuration description. Complete TODOs below and remove the _comment properties.It is recommended to use the JSON editor that supports schema hints to get more information about defined JSON properties and their description.",
```
[the remark this answers is not in the document]
Comment (vlada-shubina) [reply template_feed/Microsoft.TemplateEngine.Authoring.Templates/content/TemplateJson/template.json]: Good catch, however apparently it's only allowed on top level :(

File: template_feed/Microsoft.TemplateEngine.Authoring.Templates/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+## The templates for template authoring
+
+The package contains the templates useful for the template authoring:
+| Template name | Short name | Description|
+|---|---|---|
+|`template.json` configuration file|`template.json`|A template for template.json configuration file for .NET template.|
+|Template Package|`template-package`|A project for creating template package containing .NET templates.|
+
+The package is available for download from nuget.org.
+Please feel to contribute or provide the feedback in discussions or via opening the issue in dotnet/templating repo.
```
[the remark this answers is not in the document]
Comment (vlada-shubina) [reply template_feed/Microsoft.TemplateEngine.Authoring.Templates/README.md:10]: removed other one

**Review by JanKrivanek — APPROVED** [review]
Overal looks good!

I have couple minor comments - but those are just ideas

File: template_feed/Microsoft.TemplateEngine.Authoring.Templates/content/TemplatePackage/New.Template.Package.csproj [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    <PackageVersion>1.0</PackageVersion>
+    <Title>TODO: fill the package name here</Title>
+    <Authors>TODO: fill the author name or organization here</Authors>
+    <Description>TODO: fill the package description here</Description>
+    <PackageTags>TODO: fill the tags here</PackageTags>
+    <PackageProjectUrl>TODO: include a link to an associated project, repository, or company website</PackageProjectUrl>
+
+    <!-- Keep package type as 'Template' to show the package as template package on nuget.org and make you template available in dotnet new search.-->
+    <PackageType>Template</PackageType>
+    <TargetFramework>net7.0</TargetFramework>
```
[the remark this answers is not in the document]
Comment (vlada-shubina) [reply template_feed/Microsoft.TemplateEngine.Authoring.Templates/content/TemplatePackage/New.Template.Package.csproj:16]: https://github.com/dotnet/templating/pull/5811 also is not yet merged.

**Review by baronfel — APPROVED** [review]

File: template_feed/Microsoft.TemplateEngine.Authoring.Templates/content/TemplatePackage/New.Template.Package.csproj (hunk 3 above)
[the remark this answers is not in the document]
Comment (vlada-shubina) [reply template_feed/Microsoft.TemplateEngine.Authoring.Templates/content/TemplatePackage/New.Template.Package.csproj:16]: okay, it was merged via other PR, then makes sense - will add later.

File: template_feed/Microsoft.TemplateEngine.Authoring.Templates/content/TemplateJson/template.json [hunk 4]
```diff
@@ -0,0 +1,69 @@
+{
+  "$schema": "http://json.schemastore.org/template",
+  "$comment": "See https://aka.ms/template-json-reference for complete configuration description. Complete TODOs below and remove the $comment properties. It is recommended to use the JSON editor that supports schema hints to get more information about defined JSON properties and their description.",
+  "author": "TODO: fill author name here.",
+  "classifications": [
+    "TODO: fill classification here. Common classifications are: Library, Test, Web etc. Each classification should be a separate element of the array."
```
[the remark this answers is not in the document]
Comment (vlada-shubina) [reply template_feed/Microsoft.TemplateEngine.Authoring.Templates/content/TemplateJson/template.json]: Added, though classifications description in the doc is not great :(

File: template_feed/Microsoft.TemplateEngine.Authoring.Templates/content/TemplatePackage/New.Template.Package.csproj (hunk 3 above)
[the remark this answers is not in the document]
Comment (vlada-shubina) [reply template_feed/Microsoft.TemplateEngine.Authoring.Templates/content/TemplatePackage/New.Template.Package.csproj]: updated

File: test/Microsoft.TemplateEngine.Authoring.Templates.IntegrationTests/AuthoringTemplatesTests.cs [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+
+            TemplateVerifierOptions options = new TemplateVerifierOptions(templateName: templateShortName)
+            {
+                TemplatePath = templateLocation,
+                SnapshotsDirectory = "Snapshots",
+                OutputDirectory = workingDir,
+                DoNotPrependCallerMethodNameToScenarioName = true,
+                ScenarioName = "Basic",
+            }
+            .WithInstantiationThroughTemplateCreatorApi(new Dictionary<string, string?>());
```
[the remark this answers is not in the document]
Comment (vlada-shubina) [reply test/Microsoft.TemplateEngine.Authoring.Templates.IntegrationTests/AuthoringTemplatesTests.cs:119]: i'm fan of KISS in tests. 
Framework is changed once a year, and it's ok to update snapshots when it happens - there were only few adjustments needed.

**Pull request merged**